### DISCUSSION TITLE
Is there a way to force a decimal separator for useNumberField/NumberField?

### DISCUSSION THREAD

**Discussion by ldvilchez** [adobe/react-spectrum#5422] [Q&A] [answered]
I'm currently using `useNumberField` for our numeric fields (planning to migrate to NumberField) but I'm having some issues with a couple of fields that are used for coordinates, I need to force the decimal separator to always be a dot `.` since that's how coordinates are stored/displayed. 

I checked the documentation for the hook, component and also the MDN documentation for Intl.NumberFormat but couldn't find a definitive answer.

**Comment by snowystinger** [comment, marked as answer]
Given that lat/long coordinates really shouldn't be localized, as near as I can tell https://ux.stackexchange.com/questions/85702/localizing-coordinates

You could pass the locale `'en-US'` to the hook. Then it would always have a decimal and shouldn't cause a problem because it shouldn't be localized to anything else.

When you move to the component, you can wrap it in the i18n provider https://react-spectrum.adobe.com/react-aria/internationalization.html

**Reply by ldvilchez** [reply]
Thanks, this helped with displaying the initial value, before it was formatting the initial coordinates with commas, but I'm still having an issue with the automatic formatting.

Say I type `-33123456` it gets automatically formatted to `-33,123,456`

**Reply by snowystinger** [reply]
is that really a valid lat/long? can't they only be -90 to 90 and -180 to 180?

**Reply by ldvilchez** [reply]
Not really valid, but I was just wondering if it was possible to format to `-33.123456`.

**Reply by snowystinger** [reply]
I'm not sure how we'd know if it was 33, 3, .3, .03 etc

**Reply by ldvilchez** [reply]
Fair enough, probably asking too much. 😅 Thanks!

**Reply by ldvilchez** [reply]
Sorry to post again, but I noticed it's rounding up numbers ending with zeros, say for example `7.540000` get's trimmed to `7.54`. I thought formatting the value myself would be enough but it seems to get overrided. 🥲

**Reply by snowystinger** [reply]
Should be able to specify a minimum number of decimal places https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#minimumfractiondigits

A few things to keep in mind, if you don't need the step value, don't need localization to anything other than English, or need more decimal places than the formatter can handle, then you might be better served by a text field with your own validation. These numbers seem to be quite a special case.

**Reply by snowystinger** [reply]
I had a look at how far you actually need to represent lat/long in terms of fraction digits https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude

We support up to ~18 fraction digits (thanks JS), so I think you'll be fine.

https://codesandbox.io/s/vigorous-fast-p5p28w?file=/src/App.js

**Reply by ldvilchez** [reply]
Thank you! Yeah, I was thinking that using TextField with my own validations would be more ideal, I'll work on that.
